### PR TITLE
fix(taskframework): cannot rollback stop when destroy executor failed

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

module-taskframework

#### What this PR does / why we need it:
because stop remote task and destroy executor is two phases, stop remote action cannot be rollback if destroy failed, we should do check conditional about executor can be destroy

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/K8sJobCaller.java [hunk 1]
```diff
@@ -85,6 +85,32 @@ protected void doDestroy(JobIdentity ji, ExecutorIdentifier ei) throws JobExcept
...
+        }
+        if (k8sJobResponse.isPresent()) {
+            if (PodStatus.PENDING == PodStatus.of(k8sJobResponse.get().getResourceStatus())) {
+                JobConfiguration jobConfiguration = JobConfigurationHolder.getJobConfiguration();
+                JobEntity jobEntity = jobConfiguration.getTaskFrameworkService().find(ji.getId());
+                if ((System.currentTimeMillis() - jobEntity.getStartedTime().getTime()) / 1000 <= podConfig
+                        .getPodPendingTimeoutSeconds()) {
+                    // Pod cannot be deleted when pod pending is not timeout,
+                    // so throw exception representative cannot delete
+                    log.warn("Cannot destroy pod, pending is not timeout, jodId={0}, identifier={1}, podStatus={2}",
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/K8sJobCaller.java:105]: `{0}` may contains no position number ?

Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/K8sJobCaller.java]: okey, done

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by yizhouxw — CHANGES_REQUESTED** [review]
may change target branch to dev/4.3.0

**Comment by krihy** [comment]
> may change target branch to dev/4.3.0

okey , create a new  [PR](https://github.com/oceanbase/odc/pull/2755), and close this pr

**Comment by krihy** [comment]
see new pr: https://github.com/oceanbase/odc/pull/2745

**Pull request closed without merging**